aws/services/ec2/resource.ftl

```diff
@@ -672,6 +672,49 @@
     ]
 [/#function]
 
+[#function getInitConfigSSHPublicKeys SSHPublicKeys linkEnvironment ignoreErrors=false priority=9 ]
+    [#local SSHPublicKeysContent = "" ]
+    [#list SSHPublicKeys as id,publicKey ]
+        [#if (linkEnvironment[publicKey.SettingName])?has_content ]
+            [#local SSHPublicKeysContent += linkEnvironment[publicKey.SettingName] + " " + id ]
+            [#if (SSHPublicKeys?keys)?seq_index_of(id) != ((SSHPublicKeys?keys)?size - 1)]
+                [#local SSHPublicKeysContent += "\n"]
+            [/#if]
+        [/#if]
+    [/#list]
+    [#return
+        {
+            "${priority}_scripts" : {
+                "files" :{
+                    "/home/ec2-user/.ssh/authorized_keys2" : {
+                        "content" : {
+                            "Fn::Join" : [
+                                "",
+                                [
+                                    SSHPublicKeysContent
+                                ]
+                            ]
+                        },
+                        "mode" : "000600",
+                        "group" : "ec2-user",
+                        "owner" : "ec2-user"
+                    }
+                },
+                "commands": {
+                    "01UpdateSSHDConfig" : {
+                        "command" : "sed -i 's#^\\(AuthorizedKeysFile.*$\\)#\\1 .ssh/authorized_keys2#' /etc/ssh/sshd_config",
+                        "ignoreErrors" : ignoreErrors
+                    },
+                    "02RestartSSHDService" : {
+                        "command" : "sudo service sshd restart",
```

Comment:
Is the sudo required? I thought CFN init ran as root.

Reply:
fixed!